### ISSUE TITLE
Autofill management

### ISSUE THREAD

**Issue by ukandrewc** [MicrosoftEdge/WebView2Feedback#1261]
Do we have any way to manage saved autofill/passwords? can't see anything in the docs.

Thanks

[AB#33208906](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/33208906)

[AB#33209255](https://microsoft.visualstudio.com/90b2a23c-cab8-4e7c-90e7-a977f32c1f5d/_workitems/edit/33209255)

**Comment by maurawinstanley** [comment]
@ukandrewc We are working on a clear browsing data API in which developers will be able to pass data types they would like to clear. It is not yet released, and we hope to do so soon. Currently we do not have autofill/password management capability. In the meantime to delete saved autofill/passwords please delete the User Data folder.

**Comment by ukandrewc** [comment]
Come on guys, that's not a solution. It's a bit insulting to the history of MS, that you would even consider that as a response.

If we have 100 passwords stored, and one needs to be changed, your think we should delete all of them and start again?

Surely you can see that without edit capability, we'd all have to code our own?

**Comment by maurawinstanley** [comment]
We aren't exposing granular autofill/password clearing due to lack of bandwidth at the moment. We are taking an iterative approach by first providing the autofill API, the clear browsing data API following soon, with the potential for more in depth form management later. Given the choice between waiting to release the autofill API in conjunction with the clear browsing data API we decided to expose the autofill API in the latest release for applicable use cases. We understand this is a useful/practical ask and are looking into it.

**Comment by ukandrewc** [comment]
@maurawinstanley OK, thanks for the explanation. Do you mean developer bandwidth?

I appreciate the iterative approach, but our application is stalled because we were waiting for the promised autofill API.

Would it be very difficult to just raise an event, when password data is needed for a page, and when it needs to be stored (similar to download API)?

**Comment by maurawinstanley** [comment]
Yes, we do mean developer/team bandwidth.  We can look into what would be needed to raise an event for those scenarios.  In the meantime please feel free to give feedback on the clear browsing data API in relation to autofill mentioned [here](https://github.com/MicrosoftEdge/WebView2Feedback/issues/374)

**Comment by ukandrewc** [comment]
@maurawinstanley Just to be clear, I wasn't expecting autofill data to be stored by WebView2. If store and retrieve events are raised, they would save and retrieve autofill data, and cancel internal storage.

**Comment by maurawinstanley** [comment]
@ukandrewc Thanks for clarifying, we'll look into it!

**Comment by maurawinstanley** [comment]
@ukandrewc We will be going through our backlog soon and will update you!

**Comment by ukandrewc** [comment]
@maurawinstanley Thank you, PS, how much grovelling does it take, to get it bumped? ;-)

**Comment by mikkelmogensen** [comment]
@maurawinstanley @champnic
the ability to manage the autofill data (passwords, addresses, credit cards), especially passwords, would be very much appreciated
